added net4.8 support

ankitdas13 opened the pull request:
### Added Support for .net 4.8

shwatang commented on Razorpay.nuspec:
```diff
@@ -22,5 +22,6 @@
     <file src="net40/bin/Release/Razorpay.dll" target="lib/net40/Razorpay.dll" />
     <file src="net45/bin/Release/Razorpay.dll" target="lib/net45/Razorpay.dll" />
     <file src="net47/bin/Release/Razorpay.dll" target="lib/net47/Razorpay.dll" />
+	  <file src="net48/bin/Release/Razorpay.dll" target="lib/net48/Razorpay.dll" />
```
Indentation fix.

shwatang commented on net48/Properties/AssemblyInfo.cs:
```diff
@@ -0,0 +1,36 @@
+// General Information about an assembly is controlled through the following
+// set of attributes. Change these attribute values to modify the information
+// associated with an assembly.
+[assembly: AssemblyTitle("RazorpayClient48")]
+[assembly: AssemblyDescription("")]
+[assembly: AssemblyConfiguration("")]
+[assembly: AssemblyCompany("")]
+[assembly: AssemblyProduct("RazorpayClient48")]
+[assembly: AssemblyCopyright("Copyright ©  2024")]
```
```suggestion
[assembly: AssemblyTitle("Razorpay")]
[assembly: AssemblyDescription("Razorpay SDK for .Net")]
[assembly: AssemblyConfiguration("")]
[assembly: AssemblyCompany("Razorpay Software Private Limited")]
[assembly: AssemblyProduct("Razorpay.API")]
[assembly: AssemblyCopyright("Copyright © 2015 - 2024")]
```
Should't it be this? Correct me if mistaken.

ankitdas13 replied:
updated 

https://github.com/razorpay/razorpay-dot-net/blob/10148cd90cfb84ff83d6f4e8643488f5fe22eef5/net48/Properties/AssemblyInfo.cs#L8

shwatang commented on net48/Properties/AssemblyInfo.cs:
```diff
@@ -0,0 +1,36 @@
+//      Build Number
+//      Revision
+//
+// You can specify all the values or you can default the Build and Revision Numbers
+// by using the '*' as shown below:
+// [assembly: AssemblyVersion("1.0.*")]
```
Can we remove this?

shwatang commented on net48/packages.config:
```diff
@@ -0,0 +1,5 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<packages>
+  <package id="Newtonsoft.Json" version="13.0.3" targetFramework="net48" />
+  <package id="Portable.BouncyCastle" version="1.8.0" targetFramework="net48" />
```
why changed it from 1.9.0 version?

ankitdas13 replied:
updated 

https://github.com/razorpay/razorpay-dot-net/blob/10148cd90cfb84ff83d6f4e8643488f5fe22eef5/net48/packages.config#L4

shwatang commented on net48/RazorpayClient48.csproj:
```diff
@@ -0,0 +1,199 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="15.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
```
why we not adding DefaultTargets key?

ankitdas13 replied:
added 

https://github.com/razorpay/razorpay-dot-net/blob/10148cd90cfb84ff83d6f4e8643488f5fe22eef5/net48/RazorpayClient48.csproj#L2

shwatang commented on net48/RazorpayClient48.csproj:
```diff
@@ -0,0 +1,199 @@
+    <AssemblyName>Razorpay</AssemblyName>
+    <TargetFrameworkVersion>v4.8</TargetFrameworkVersion>
+    <FileAlignment>512</FileAlignment>
+    <Deterministic>true</Deterministic>
```
What this does?

ankitdas13 replied:
removed

shwatang commented on net48/RazorpayClient48.csproj:
```diff
@@ -0,0 +1,199 @@
+  </PropertyGroup>
+  <ItemGroup>
+    <Reference Include="crypto, Version=1.8.0.0, Culture=neutral, PublicKeyToken=0e99375e54769942, processorArchitecture=MSIL">
+      <HintPath>packages\Portable.BouncyCastle.1.8.0\lib\net46\crypto.dll</HintPath>
+    </Reference>
+    <Reference Include="Newtonsoft.Json, Version=13.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed, processorArchitecture=MSIL">
+      <HintPath>packages\Newtonsoft.Json.13.0.3\lib\net45\Newtonsoft.Json.dll</HintPath>
```
Please once check the HintPath and confirm if its correct.

ankitdas13 replied:
its correct

![Screenshot 2024-06-20 153132](https://github.com/razorpay/razorpay-dot-net/assets/89454448/bf13fb1a-12a9-40af-8bf4-25762c90628e)

shwatang commented on net48/RazorpayClient48.csproj:
```diff
@@ -0,0 +1,199 @@
+      <HintPath>packages\Newtonsoft.Json.13.0.3\lib\net45\Newtonsoft.Json.dll</HintPath>
+    </Reference>
+    <Reference Include="System" />
+    <Reference Include="System.IO.FileSystem, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a, processorArchitecture=MSIL">
+      <HintPath>packages\System.IO.FileSystem.4.0.0\lib\net46\System.IO.FileSystem.dll</HintPath>
+      <Private>True</Private>
+      <Private>True</Private>
+    </Reference>
+    <Reference Include="System.IO.FileSystem.Primitives, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a, processorArchitecture=MSIL">
+      <HintPath>packages\System.IO.FileSystem.Primitives.4.0.0\lib\net46\System.IO.FileSystem.Primitives.dll</HintPath>
+      <Private>True</Private>
+      <Private>True</Private>
```
Why including this as part this PR?

ankitdas13 replied:
to use `System.Net.HttpWebRequest` namespace in net4.8 we need to add reference.

shwatang reviewed (commented):
Please address the comments. Also, please do confirm if this change has been tested thoroughly.

ankitdas13 commented on net48/RazorpayClient48.csproj:
```diff
@@ -0,0 +1,199 @@
+    <StartupObject />
+  </PropertyGroup>
+  <ItemGroup>
+    <Reference Include="crypto, Version=1.8.0.0, Culture=neutral, PublicKeyToken=0e99375e54769942, processorArchitecture=MSIL">
```
updated

https://github.com/razorpay/razorpay-dot-net/blob/10148cd90cfb84ff83d6f4e8643488f5fe22eef5/net48/RazorpayClient48.csproj#L36

ankitdas13 commented:
> Please address the comments. Also, please do confirm if this change has been tested thoroughly.

I addressed these comments and made changes, as well as testing it

shwatang reviewed (approved):
LGTM
